src/extension-runners/firefox-android.js

```diff
@@ -0,0 +1,653 @@
+      params: {
+        adb,
+        firefoxApk,
+      },
+    } = this;
+    // Discovery and select a Firefox for Android version.
+    const pmList = await adbClient.shell(selectedAdbDevice.id, [
+      'pm', 'list', 'packages',
+    ]).then(adb.util.readAll);
+
+    const packages = pmList.toString().split('\n')
+                           .map((line) => line.replace('package:', '').trim())
+                           .filter((line) => {
+                             return (
+                               line.startsWith('org.mozilla.fennec') ||
+                               line.startsWith('org.mozilla.firefox')
+                             );
+                           });
+
+    if (packages.length === 0) {
+      throw new UsageError(
+        'No Firefox packages found of the selected Android device');
+    }
+
+    const pkgsListMsg = (pkgs) => {
+      return pkgs.map((pkg) => ` - ${ pkg}`).join('\n');
+    };
+
+    if (!firefoxApk) {
+      log.info(`\nPackages found:\n${pkgsListMsg(packages)}`);
+      throw new UsageError('Select one of the packages using --firefox-apk');
+    }
+
+    const filteredPackages = packages.filter((line) => line === firefoxApk);
+
+    if (filteredPackages.length === 0) {
+      const pkgsList = pkgsListMsg(filteredPackages);
+      throw new UsageError(`Package not found: ${firefoxApk} in ${pkgsList}`);
+    }
+
+    this.selectedFirefoxApk = filteredPackages[0];
+    log.info(`Selected Firefox for Android APK: ${this.selectedFirefoxApk}`);
+  }
+
+  async adbForceStopSelectedPackage() {
+    const {
+      adbClient,
+      selectedAdbDevice,
+      selectedFirefoxApk,
+      params: {
+        adb,
+      },
+    } = this;
+    log.info(`Stop any existent instance of ${selectedFirefoxApk}...`);
+    await adbClient.shell(selectedAdbDevice.id, [
+      'am', 'force-stop', selectedFirefoxApk,
+    ]).then(adb.util.readAll);
+  }
+
+  async adbPrepareProfileDir() {
+    const {
+      adbClient,
+      selectedAdbDevice,
+      selectedFirefoxApk,
+      params: {
+        firefoxApp,
+      },
+    } = this;
+    // Create the preferences file and the Fennec temporary profile.
+    log.info(`Preparing a temporary profile for ${selectedFirefoxApk}...`);
+    const profile = await firefoxApp.createProfile({app: 'fennec'});
+
+    const deviceProfileDir = this.getDeviceProfileDir();
+
+    await adbClient.shell(selectedAdbDevice.id, [
+      'mkdir', '-p', deviceProfileDir,
+    ]);
+    await adbClient.push(selectedAdbDevice.id, `${profile.profileDir}/user.js`,
+                         `${deviceProfileDir}/user.js`);
+    log.debug(`Created temporary profile at ${deviceProfileDir}.`);
+  }
+
+  async adbStartSelectedPackage() {
+    const {
+      adbClient,
+      selectedFirefoxApk,
+      selectedAdbDevice,
+    } = this;
+
+    const deviceProfileDir = this.getDeviceProfileDir();
+
+    log.info(
+      `Starting ${selectedFirefoxApk} on the profile ${deviceProfileDir}...`
```

Comment:
I think it's not so important to tell the user what profile it is so this could be split into two messages:
```js
log.info(`Starting ${selectedFirefoxApk}...`); 
log.debug(`Using profile ${deviceProfileDir}`);
```